### PR TITLE
REST API tests for GET jobs + GET jobs/ID/annotations

### PR DESCRIPTION
**Pull request by nmanovic**
### Checklist

- [x] I submit my changes into the `develop` branch
- [ ] I have added a description of my changes into [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
- [ ] I have updated the [documentation](
  https://github.com/opencv/cvat/blob/develop/README.md#documentation) accordingly
- [x] I have added tests to cover my changes
- [ ] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [x] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2022 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

**Comment by nmanovic** [comment]
@kirill-sizov , I will merge the PR to unblock others. Please review it and share your comments next week.

File: tests/rest_api/test_0100_jobs.py [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+            job_staff = get_job_staff(job, tasks, projects)
+            jid = str(job['id'])
+
+            for user in users:
+                if user['id'] in job_staff | org_staff:
+                    self._test_get_job_annotations_200(user['username'],
+                        jid, annotations['job'][jid], **kwargs)
+                else:
+                    self._test_get_job_annotations_403(user['username'],
+                        jid, **kwargs)
```
Comment (kirill-sizov) [inline tests/rest_api/test_0100_jobs.py:170]: I see that this test can do more than one test for a similar test case, for example for test case (worker, not job staff) the function `_test_get_job_annotations_403` will be call more than once. I think from IAM testing point of view we have no reason in testing similar action for all users with same privilege/memberships. What do you think, @nmanovic?

Comment (nmanovic) [reply tests/rest_api/test_0100_jobs.py:170]: @kirill-sizov , I agree. If you see an easy way to reduce redundancy, I'm OK with the approach.

File: tests/rest_api/test_0100_jobs.py [hunk 2]
```diff
@@ -0,0 +1,170 @@
+# Copyright (C) 2021 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
```
Comment (kirill-sizov) [inline tests/rest_api/test_0100_jobs.py:3]: The name of this file led me to think why do we use numbers in filenames? If these numbers are not needed then I suggest to follow pattern `test_{resource_name}.py` or if we have some reasons to keep this numbers in filenames than it's better to describe (in README) a convention of assignment of this numbers.

**Pull request merged**